packages/remote-replay-launcher/src/upload-assets-and-trigger-test-run.ts

```diff
@@ -18,6 +20,50 @@ import {
   ExecuteRemoteTestRunResult,
 } from "./types";
 
+const POLL_FOR_BASE_TEST_RUN_INTERVAL_MS = 10_000;
+const POLL_FOR_BASE_TEST_RUN_MAX_TIMEOUT_MS = 5 * 60 * 1000; // 5 minutes
+
+/**
+ * Polls completeAssetUpload every 10 seconds until result.testRun exists or timeout is reached.
+ */
+export const tryCompleteAssetUpload = async (
+  completeAssetUploadArgs: Parameters<typeof completeAssetUpload>[0]
+): Promise<{ testRun: TestRun | null }> => {
+  const startTime = Date.now();
+  let result = await completeAssetUpload(completeAssetUploadArgs);
+  let testRun = result?.testRun;
+  let baseNotFound = result?.baseNotFound;
+  while (!testRun && baseNotFound) {
+    if (Date.now() - startTime > POLL_FOR_BASE_TEST_RUN_MAX_TIMEOUT_MS) {
+      throw new Error(
+        `Timed out after ${
+          POLL_FOR_BASE_TEST_RUN_MAX_TIMEOUT_MS / 1000
+        } seconds waiting for test run`
+      );
+    }
+    await new Promise((resolve) =>
+      setTimeout(resolve, POLL_FOR_BASE_TEST_RUN_INTERVAL_MS)
+    );
+    result = await triggerRunOnDeployment(completeAssetUploadArgs);
+    testRun = result?.testRun;
+    baseNotFound = result?.baseNotFound;
+  }
```

Comment:
We should have some logging (maybe not every time we go around the loop, but at least when we start waiting and once every ~30s so it doesn't look like things are stuck. Otherwise users might cancel the workflow.